src/groovy/org/transmartproject/db/dataquery/clinical/AcrossTrialsDataQuery.groovy

```diff
@@ -57,31 +59,29 @@ class AcrossTrialsDataQuery {
             throw new IllegalStateException('init() not called successfully yet')
         }
 
-        def condition = new ChoppedInQueryCondition('patient.id', patientIds)
-        Query query = session.createQuery """
-                SELECT
-                    patient.id,
-                    modifierCd,
-                    valueType,
-                    textValue,
-                    numberValue
-                FROM ObservationFact fact
-                WHERE
-                    ${condition.queryConditionTemplate}
-                AND
-                    fact.modifierCd IN (:modifierCds)
-                ORDER BY
-                    patient ASC,
-                    modifierCd ASC"""
-
-        query.cacheable = false
-        query.readOnly  = true
-        query.fetchSize = FETCH_SIZE
-
-        condition.parametersValues.each { parVal -> query.setParameterList parVal.key, parVal.value }
-        query.setParameterList 'modifierCds', clinicalVariables*.modifierCode
-
-        query.scroll ScrollMode.FORWARD_ONLY
+        def criteriaBuilder = createCriteriaBuilder(ObservationFact, 'obs', session)
+        criteriaBuilder.with {
+            projections {
+                property 'patient.id'
+                property 'modifierCd'
+                property 'valueType'
+                property 'textValue'
+                property 'numberValue'
+            }
+            order 'patient.id'
+            order 'modifierCd'
+        }
+
+        if (patients instanceof DetachedCriteria) {
+            criteriaBuilder.add(getHibernateInCriterion('patient.id',
+                    ((DetachedCriteria) patients).where { projections { id() } }))
+        } else {
+            criteriaBuilder.in('patient',  Lists.newArrayList(patients))
+        }
+
+        criteriaBuilder.in('modifierCd', clinicalVariables*.code)
+
+        criteriaBuilder.scroll ScrollMode.FORWARD_ONLY
```

Comment:
This doesn't set `cacheable`, `readOnly` or `fetchSize` on the query. At least `fetchSize` I remember it's essential to prevent the whole result set from being loaded into memory.